qiskit/synthesis/evolution/suzuki_trotter.py

```diff
@@ -84,10 +73,12 @@ def __init__(
                 Pauli string. Per default, a single Pauli evolution is decomposed in a CX chain
                 and a single qubit Z rotation.
         """
-        # TODO replace deprecation warning by the following error and add unit test for odd
-        # if order % 2 == 1:
-        #   raise ValueError("Suzuki product formulae are symmetric and therefore only defined "
-        #                  "for even orders.")
+
+        if order % 2 == 1:
+            raise ValueError(
```

Comment:
Please, extend the documentation with `Raises` for those even `order` cases.

Reply:
Will add in next commit.